kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/BaseOperation.java

```diff
@@ -1011,4 +1012,9 @@ public static URL appendListOptionParams(URL base, ListOptions listOptions) {
     return urlBuilder.build();
   }
 
+  @Override
+  public Stream<R> resources() {
+    return list().getItems().stream().map(this::resource);
```

Comment:
What's the purpose of the map/peek? It will set the last listed item as the current item.

Reply:
map does not perform a peek, it will convert from a stream of HasMetadata to a stream of Resources

Reply:
OK, GH interface is playing dirty again.